lib/nx_signal/filters.ex

```diff
@@ -4,78 +4,92 @@ defmodule NxSignal.Filters do
   """
   import Nx.Defn
 
-  @pi :math.pi()
-
   @doc ~S"""
-  Calculates the normalized sinc function $sinc(t) = \frac{sin(\pi t)}{\pi t}$
+  Performs a median filter on a rank 1 or rank 2 tensor.
 
-  ## Examples
+  ## Options
 
-      iex> NxSignal.Filters.sinc(Nx.tensor([0, 0.25, 1]))
-      #Nx.Tensor<
-        f32[3]
-        [1.0, 0.9003162980079651, -2.7827534054836178e-8]
-      >
+    * `:kernel_shape` - the shape of the sliding window.
+    It must be compatible with the shape of the tensor.
   """
   @doc type: :filters
-  defn sinc(t) do
-    t = t * @pi
-    zero_idx = Nx.equal(t, 0)
+  def median(t = %Nx.Tensor{shape: {length}}, opts) do
+    validate_median_opts!(t, opts)
+    {kernel_length} = opts[:kernel_shape]
+    median(Nx.reshape(t, {1, length}), kernel_shape: {1, kernel_length})
+  end
 
-    # Define sinc(0) = 1
-    Nx.select(zero_idx, 1, Nx.sin(t) / t)
+  def median(t = %Nx.Tensor{shape: {_h, _w}}, opts) do
+    validate_median_opts!(t, opts)
+    median_n(t, opts)
   end
 
-  defn median_filter(t, opts \\ [kernel_size: 3]) do
+  def median(_t, _opts), do: raise(ArgumentError, message: "tensor must be of rank 1 or 2")
+
+  defnp median_n(t, opts) do
+    kernel_shape = opts[:kernel_shape]
+
     {height, width} = Nx.shape(t)
-    kernel_tensor = Nx.broadcast(0.0, {opts[:kernel_size]})
+
+    kernel_tensor = Nx.broadcast(0.0, kernel_shape)
     output = Nx.broadcast(0.0, Nx.shape(t))
 
-    {_, _, _, _, _, result} =
-      while {i = 0, t, kernel_tensor, height, width, output}, i < height do
+    {result, _} =
+      while {output, {i = 0, t, kernel_tensor, height, width}}, i < height do
         row = Nx.broadcast(0.0, {elem(Nx.shape(t), 1)})
 
-        {_, _, _, _, _, ith_row} =
-          while {j = 0, t, i, kernel_tensor, width, row}, j < width do
-            ij_median =
-              ij_window_median(t, i, j, kernel_tensor)
+        {ith_row, _} =
+          while {row, {j = 0, t, i, kernel_tensor, width}}, j < width do
+            median =
+              window_median(t, i, j, kernel_tensor)
               |> Nx.broadcast({1})
 
-            {j + 1, t, i, kernel_tensor, width, Nx.put_slice(row, [j], ij_median)}
+            {Nx.put_slice(row, [j], median), {j + 1, t, i, kernel_tensor, width}}
           end
 
-        {i + 1, t, kernel_tensor, height, width,
-         Nx.put_slice(output, [i, 0], Nx.stack(ith_row, axis: 0))}
+        {Nx.put_slice(output, [i, 0], Nx.stack(ith_row, axis: 0)),
+         {i + 1, t, kernel_tensor, height, width}}
       end
 
-    result
+    Nx.squeeze(result)
   end
 
-  defnp ij_window_median(t, i, j, kernel_tensor) do
-    {kernel_size} = Nx.shape(kernel_tensor)
-    window_pad_size = Nx.round((kernel_size - 1) / 2)
+  defnp window_median(t, i, j, kernel_tensor) do
+    kernel_shape = Nx.shape(kernel_tensor)
+    kernel_dims = [elem(kernel_shape, 0), elem(kernel_shape, 1)]
```

Comment:
```suggestion
    {k0, k1} = Nx.shape(kernel_tensor)
    kernel_dims = [k0, k1]
```

And then use `k0` and `k1` below too.

Also, if you use `deftransformp`, you'll be able to calculate the paddings using  `div` and keep them as numbers